### PR TITLE
Add acceptance test to verify collection of secrets from secretRef

### PR DESCRIPTION
**Pull request by pratikjagrut**
### Motivation
https://github.com/redhat-developer/service-binding-operator/issues/808
https://issues.redhat.com/browse/APPSVC-785

### CODE REVIEW

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 1]
```diff
@@ -469,3 +469,66 @@ Feature: Bind values from a secret referred in backing service resource
         Then Service Binding "ssa-3" is ready
         And The application env var "BACKEND_USERNAME" has value "AzureDiamond"
         And The application env var "BACKEND_PASSWORD" has value "hunter2"
+
+    # Remove this disable tag once this issue is closed: https://github.com/redhat-developer/service-binding-operator/issues/808
+    @disabled
```
Comment (pedjak) [inline test/acceptance/features/bindAppToServiceUsingSecret.feature:474]: we should see this failing in some CI run.

**Review by pedjak — CHANGES_REQUESTED** [review]

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 2]
```diff
@@ -469,3 +469,66 @@ Feature: Bind values from a secret referred in backing service resource
...
+          metadata:
+              name: sb-get-secret-data
+          spec:
+              services:
+              - group: apps
+                version: v1
+                kind: Deployment
+                name: mysql-db-x
+          """
+        Then jq ".status.conditions[] | select(.type=="CollectionReady").status" of Service Binding "sb-get-secret-data" should be changed to "True"
```
[the remark this answers is not in the document]
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToServiceUsingSecret.feature:530]: Okay

File: test/acceptance/features/bindAppToServiceUsingSecret.feature [hunk 3]
```diff
@@ -469,3 +469,66 @@ Feature: Bind values from a secret referred in backing service resource
...
+            stringData:
+                MYSQL_DATABASE: "mydb"
+                MYSQL_ROOT_PASSWORD: "mydbpass"
+                MYSQL_USER: "mysql"
+                MYSQL_PASSWORD: "mysqlpass"
+            """
+        * The deployment is present
+            """
+            apiVersion: apps/v1
+            kind: Deployment
```
[the remark this answers is not in the document]
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToServiceUsingSecret.feature:491]: okay

File: test/acceptance/features/bindAppToServiceUsingSecret.feature (hunk 1 above)
Comment (pratikjagrut) [reply test/acceptance/features/bindAppToServiceUsingSecret.feature:474]: okay.

**Comment by pratikjagrut** [comment]
This test fails as expected [CI link](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-developer_service-binding-operator/809/pull-ci-redhat-developer-service-binding-operator-master-4.6-acceptance/1340946906024513536)

**Pull request merged**